File: Som/Data/CompletelyRandomDataProvider.cs
using System;
using System.Collections.Generic;

namespace Som.Data
{
    public class CompletelyRandomDataProvider : ILearningDataProvider
    {
        public int D { get; private set; }
        private int _vectorsCountToReturn;

        public Random Random { get; private set; }
        public CompletelyRandomDataProvider(int dimentionSize, int vectorsCountToReturn)
        {
            D = dimentionSize;
            _vectorsCountToReturn = vectorsCountToReturn;
            Random = new Random();
        }

        public int LearningVectorsCount
        {
            get { return _vectorsCountToReturn; }
        }

        public int DataVectorDimention
        {
            get { return D; }
        }

        public double[] GetLearingDataVector(int vectorIndex)
        {
            var result = new double[D];
            for (int i = 0; i < D; i++)
            {
                result[i] = Random.NextDouble();
            }
            return result;
        }
    }
}

File: JavaSom/Data/CompletelyRandomDataProvider.java
package Data;

import java.util.Random;

public class CompletelyRandomDataProvider implements ILearningDataProvider
	{
		private int privateD;
		public final int getD()
		{
			return privateD;
		}
		private void setD(int value)
		{
			privateD = value;
		}
		private int _vectorsCountToReturn;

		private java.util.Random privateRandom;
		public final java.util.Random getRandom()
		{
			return privateRandom;
		}
		private void setRandom(java.util.Random value)
		{
			privateRandom = value;
		}
		public CompletelyRandomDataProvider(int dimentionSize, int vectorsCountToReturn)
		{
			setD(dimentionSize);
			_vectorsCountToReturn = vectorsCountToReturn;
			setRandom(new Random());
		}

		public final int getLearningVectorsCount()
		{
			return _vectorsCountToReturn;
		}

		public final int getDataVectorDimention()
		{
			return getD();
		}

		public final double[] GetLearingDataVector(int vectorIndex)
		{
			double[] result = new double[getD()];
			for (int i = 0; i < getD(); i++)
			{
				result[i] = getRandom().nextDouble();
			}
			return result;
		}
	}